### PR TITLE
feat: basic authentication

### LINKED ISSUE #1: Basic authentication support
I did it by manually adding the headers by supplying a custom http.Client.
A dedicated option would be more convenient

```go
package logger

import (
	"encoding/base64"
	"net/http"
	"os"

	"github.com/sirupsen/logrus"
	log "github.com/sirupsen/logrus"
	"github.com/yukitsune/lokirus"
)

type AddHeaderTransport struct {
	T http.RoundTripper
}

func (adt *AddHeaderTransport) RoundTrip(req *http.Request) (*http.Response, error) {
	auth := base64.StdEncoding.EncodeToString([]byte(os.Getenv("LOKI_USERNAME") + ":" + os.Getenv("LOKI_PASSWORD")))
	req.Header.Add("Authorization", "Basic "+auth)
	return adt.T.RoundTrip(req)
}

func NewAddHeaderTransport(T http.RoundTripper) *AddHeaderTransport {
	if T == nil {
		T = http.DefaultTransport
	}
	return &AddHeaderTransport{T}
}

func SetupLokiLogger() {
	if os.Getenv("LOKI_URL") == "" {
		logrus.Warn("No Loki URL configured")
		return
	}

	opts := lokirus.NewLokiHookOptions().
		// Grafana doesn't have a "panic" level, but it does have a "critical" level
		// https://grafana.com/docs/grafana/latest/explore/logs-integration/
		WithLevelMap(lokirus.LevelMap{logrus.PanicLevel: "critical"}).
		WithHttpClient(&http.Client{
			Transport: NewAddHeaderTransport(nil),
		})

	hook := lokirus.NewLokiHookWithOpts(
		os.Getenv("LOKI_URL"),
		opts,
		logrus.InfoLevel,
		logrus.WarnLevel,
		logrus.ErrorLevel,
		logrus.FatalLevel)

	log.AddHook(hook)
}
```

### PR DESCRIPTION
**Pull request by clouedoc**
Closes #1 

Code review and criticisms are appreciated.

Do you have a clue on how I could write a nice test for this?

### CODE REVIEW

File: loki_hook_options.go [hunk 1]
```diff
@@ -57,6 +63,14 @@ type LokiHookOptions interface {
...
 	// By default, the logrus.TextFormatter will be used
 	WithFormatter(logrus.Formatter) LokiHookOptions
+
+	// WithBasicAuth allows to set a username and password to use when writing to the remote Loki host.
+	WithBasicAuth(username, password string) LokiHookOptions
+}
+
+type basicAuthCredentials struct {
+	username string
+	password string
```
Comment (YuKitsune) [inline loki_hook_options.go]: nit: Since we can return this via the `BasicAuth()` method, let's make this public

File: example/main.go [hunk 2]
```diff
@@ -18,6 +19,7 @@ func main() {
 			"app":         "example",
 			"environment": "development",
 		})
+		// .WithBasicAuth("admin", "secretpassword")
```
Comment (YuKitsune) [inline example/main.go]:
```suggestion
		.WithBasicAuth("admin", "secretpassword") // Optional
```
Should also reflect this in the readme

**Review by YuKitsune — COMMENTED** [review]
Thanks @clouedoc for the PR!
I've left some feedback for you to consider.

> Do you have a clue on how I could write a nice test for this?

I'd suggest adding a method to the `mockRoundTripper` that just returns `req.BasicAuth()`.
Example:
```go
func (r *mockRoundTripper) BasicAuth() (string, string, bool) {
    return r.req.BasicAuth()
}
```

Does that help?

**Comment by clouedoc** [comment]
> Does that help?

It did. Please tell me if the tests fit your taste.

Thanks for this nice project by the way, it works as expected.

**Comment by YuKitsune** [comment]
My pleasure, thank you for the PR!
This has been merged and released in [v1.0.1](https://github.com/YuKitsune/lokirus/releases/tag/v1.0.1) 👍

**Pull request merged**